bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/types/Type.java

```diff
@@ -31,6 +31,48 @@
  */
 public interface Type {
 
+    // TODO: remove default implementations when standard library types are updated
```

Comment:
I decided to add them to `Type` since we are dealing with `Type` in [`TypeUtils`](https://github.com/ballerina-platform/ballerina-lang/blob/fe049605d48698d100bdcd464eef80288a9b8480/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/TypeUtils.java#L154). If we move them the `BType` I think I have to do a `instanceof` check fallowed by checked cast. For that I observed our gain going down from around 8% to 6%.

Reply:
Can we cast to BType without the instanceof check? IINM, everything is a BType. Can we check the gain then?

Reply:
This didn't had a measurable effect (at least on my benchmarks) I believe this is because JVM can merge the actual runtime type check for the `instanceof` and cast together, where as for just cast it still have to do the type check